gcd/src/main/java/org/spine3/server/storage/datastore/DatastoreStorageFactory.java

```diff
@@ -165,4 +190,70 @@ protected void setDatastore(DatastoreWrapper datastore) {
         this.datastore = datastore;
     }
 
+    /**
+     * Creates new instance of {@link Builder}.
+     */
+    public static Builder newBuilder() {
+        return new Builder();
+    }
+
+    /**
+     * A builder for the {@code DatastoreStorageFactory}.
+     */
+    public static class Builder {
+
+        private Datastore datastore;
+        private boolean multitenant;
+        private ColumnTypeRegistry<DatastoreColumnType> typeRegistry = DatastoreTypeRegistry.defaultInstance();
+
+        private Builder() {
+            // Avoid direct initialization
+        }
+
+        /**
+         * @param datastore the {@link Datastore} to use for the DB interactions
+         * @return self for method chaining
+         */
+        public Builder setDatastore(Datastore datastore) {
+            this.datastore = datastore;
+            return this;
+        }
+
+        /**
+         * Switches the {@code DatastoreStorageFactory} that is being built to be or not to be multitenant.
+         *
+         * @param multitenant {@code true} if the {@code DatastoreStorageFactory} should be multitenant or not
+         * @return self for method chaining
+         */
+        public Builder setMultitenant(boolean multitenant) {
+            this.multitenant = multitenant;
```

Comment:
Can we please check for `null` values right in these setters? To follow the approach we have in `core-java`.